network/src/protocol/mod.rs

```diff
@@ -483,8 +483,8 @@ impl ProtocolHandler {
 		}
 	}
 
-	fn on_connect(&mut self, peer: PeerId, roles: Roles) {
-		let claimed_validator = roles.contains(Roles::AUTHORITY);
+	fn on_connect(&mut self, peer: PeerId, role: ObservedRole) {
+		let claimed_validator = matches!(role, ObservedRole::OurSentry | ObservedRole::OurGuardedAuthority | ObservedRole::Authority);
```

Comment:
This should be properly reviewed.
I kept the exact behaviour as it was, but maybe that's not desired, I don't know.

Reply:
For the moment, this behavior is OK. There are definitely some things that we could do better now that we are aware of our own sentry nodes and guarded validator.